starlite/controller/generic_controller.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from typing_extensions import Self, get_args, get_origin
+
+from .base import Controller
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+__all__ = ("GenericController",)
+
+T = TypeVar("T")
+
+
+class GenericController(Controller, Generic[T]):
+    """Controller type that supports generic inheritance hierarchies."""
+
+    _data_type: type[T]
+
+    def __class_getitem__(cls, data_type: type[T]) -> type[Self]:
+        return type(f"GenericController[{data_type.__name__}]", (cls,), {"_data_type": data_type})
+
+    def _get_new_args(self, args: tuple[Any, ...], origin: Any | None) -> tuple[Any, ...] | None:
+        if not args and origin:
+            return None
+
+        type_var_found = False
+        new_args = []
+        for arg in args:
+            if isinstance(arg, TypeVar):
+                new_args.append(self._data_type)
+                type_var_found = True
+                continue
+            new_args.append(arg)
+
+        if not type_var_found:
+            return None
+
+        return tuple(new_args)
+
+    @staticmethod
+    def _rebuild_annotation(original_annotation: Any, origin: Any | None, args_tuple: tuple[Any, ...]) -> Any:
```

Comment:
I don't think this method needs to be attached to the class. It can and should go into a utils module, or be declared at the top of the file

Reply:
yeh I agree - have moved to a helper